From error27@gmail.com  Wed Jan 21 09:51:00 2009
From: Dan Carpenter <error27@gmail.com>
Date: Tue, 20 Jan 2009 15:07:36 +0300 (EAT)
Subject: Staging: altpciechdma: Null deref in altpciechdma.c remove()
To: leon.woestenberg@axon.tv, heppermannwdt@gmail.com
Cc: gregkh@suse.de
Message-ID: <alpine.DEB.2.00.0901201449280.21673@bikeee>


If dev is NULL it prints an error message.  The error message dereferences
dev.

Compile tested only.

Signed-off-by: Dan Carpenter <error27@gmail.com>
Cc: Leon Woestenberg <leon.woestenberg@axon.tv>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/altpciechdma/altpciechdma.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/drivers/staging/altpciechdma/altpciechdma.c
+++ b/drivers/staging/altpciechdma/altpciechdma.c
@@ -946,7 +946,8 @@ static void __devexit remove(struct pci_
 	struct ape_dev *ape;
 	printk(KERN_DEBUG "remove(0x%p)\n", dev);
 	if ((dev == 0) || (dev->dev.driver_data == 0)) {
-		printk(KERN_DEBUG "remove(dev = 0x%p) dev->dev.driver_data = 0x%p\n", dev, dev->dev.driver_data);
+		printk(KERN_DEBUG "remove(dev = 0x%p) dev->dev.driver_data = 0x%p\n",
+			dev, (dev? dev->dev.driver_data: NULL));
 		return;
 	}
 	ape = (struct ape_dev *)dev->dev.driver_data;
